yarn/src/main/scala/org/apache/spark/deploy/yarn/ApplicationMaster.scala

```diff
@@ -346,6 +346,9 @@ private[spark] class ApplicationMaster(
                 val currentAllocationInterval =
                   math.min(heartbeatInterval, nextAllocationInterval)
                 nextAllocationInterval *= 2
+                // avoid overflow
+                nextAllocationInterval = math.min(
+                  nextAllocationInterval, ApplicationMaster.MAX_RM_HEARTBEAT_INTERVAL_MS)
```

Comment:
Can we cap at `heartbeatInterval` to avoid introducing another variable?  Or replace `nextAllocationInterval *= 2` with `nextAllocationInterval = currentAllocationInterval * 2`?